src/libraries/System.Net.NetworkInformation/src/System/Net/NetworkInformation/NetworkAddressChange.Unix.cs

```diff
@@ -140,53 +142,56 @@ public partial class NetworkChange
         private static unsafe void CreateSocket()
         {
             Debug.Assert(s_socket == null, "s_socket is not null, must close existing socket before opening another.");
-            int newSocket;
+            IntPtr newSocket;
             Interop.Error result = Interop.Sys.CreateNetworkChangeListenerSocket(&newSocket);
             if (result != Interop.Error.SUCCESS)
             {
                 string message = Interop.Sys.GetLastErrorInfo().GetErrorMessage();
                 throw new NetworkInformationException(message);
             }
 
-            s_socket = new SocketWrapper(newSocket);
+            s_socket = new Socket(new SafeSocketHandle(newSocket, ownsHandle: true));
 
-            new Thread(args => LoopReadSocket((SocketWrapper)args!))
-            {
-                IsBackground = true,
-                Name = ".NET Network Address Change"
-            }.UnsafeStart(s_socket);
+            // Don't capture ExecutionContext.
+            ThreadPool.UnsafeQueueUserWorkItem(static socket =>
+                _ = ReadEventsAsync(socket),
+                s_socket, preferLocal: false);
         }
 
         private static void CloseSocket()
         {
             Debug.Assert(s_socket != null, "s_socket was null when CloseSocket was called.");
-            Interop.Error result = Interop.Sys.CloseNetworkChangeListenerSocket(s_socket != null ? s_socket.Socket : -1);
-            if (result != Interop.Error.SUCCESS)
-            {
-                string message = Interop.Sys.GetLastErrorInfo().GetErrorMessage();
-                throw new NetworkInformationException(message);
-            }
-
+            s_socket.Dispose();
             s_socket = null;
         }
 
-        private static unsafe void LoopReadSocket(SocketWrapper initiallyCreatedSocket)
+        private static async Task ReadEventsAsync(Socket socket)
```

Comment:
This is one case where `async void` is probably desirable.  Otherwise, this is just going to silently eat any exceptions that occur.

Alternatively, if we don't want to crash the process in the case of failure, we could at least log any exceptions that occur by wrapping the whole body in a try/catch.

Reply:
I didn't dare to use `async void` :)

I'll change it. Where should I log the exceptions to?

Reply:
> I didn't dare to use async void :)

If you want to add try/catch, you don't need to use `async void` :)  I just want to avoid having the exceptions be eaten entirely... with `async Task`, we'll end up eating them by default and they'd then show up as potential noise on `TaskScheduler.UnhandledTaskException`.

> Where should I log the exceptions to?

Probably NetEventSource.Log.Error, but @MihaZupan can share if there's a better location.

Is there any other clean up we should be doing in the face of an unexpected exception?